tests/ballerina-tools-integration-test/src/test/java/org/ballerinalang/test/packaging/PackagingInitTestCase.java

```diff
@@ -0,0 +1,253 @@
+        Files.createDirectories(projectPath);
+
+        sourceFiles = new ArrayList<>();
+        packageDescFiles = new ArrayList<>();
+
+        sourceFiles.add(new SrcFile("", FileType.SERVICE));
+        packageDescFiles.add(new PackageMdFile("", FileType.SERVICE));
+
+        InitHandler.initialize(projectPath, manifest, sourceFiles, packageDescFiles);
+
+        Assert.assertEquals(Files.exists(projectPath.resolve("hello_service.bal")), true);
+        Assert.assertEquals(Files.exists(projectPath.resolve("Ballerina.toml")), true);
+        Assert.assertEquals(Files.exists(projectPath.resolve(".ballerina")), true);
+
+        // Test for service
+        ballerinaServerForService = ServerInstance.initBallerinaServer();
+        ballerinaServerForService.startBallerinaServer(projectPath.resolve("hello_service.bal").toString());
+        HttpResponse response = HttpClientRequest.doGet(ballerinaServerForService.getServiceURLHttp("hello/sayHello"));
+        Assert.assertEquals(response.getResponseCode(), 200, "Response code mismatched");
+        ballerinaServerForService.stopServer();
+    }
+
+    @Test(description = "Test running init without doing any changes on an already existing project",
+            dependsOnMethods = "testInitWithMainInPackage")
+    public void testInitOnExistingProject() throws Exception {
+        Path projectPath = tempProjectDirectory.resolve("firstTestWithPackagesMain");
+        InitHandler.initialize(projectPath, manifest, new ArrayList<>(), new ArrayList<>());
+
+        Path packagePath = projectPath.resolve("foo");
+        Assert.assertEquals(Files.exists(packagePath.resolve("main.bal")), true);
+        Assert.assertEquals(Files.exists(projectPath.resolve("Ballerina.toml")), true);
+        Assert.assertEquals(Files.exists(packagePath.resolve("tests").resolve("main_test.bal")), true);
+
+        ballerinaServerForMain = new ServerInstance(serverZipPath);
+        String[] clientArgs = {"--sourceroot", projectPath.toString(), "foo"};
+        logLeecher = new LogLeecher("Hello World!");
+        ballerinaServerForMain.addLogLeecher(logLeecher);
+        ballerinaServerForMain.runMain(clientArgs, getEnvVariables(), "run");
+    }
+
+    @Test(description = "Test running init on an already existing project and create a new package",
+            dependsOnMethods = "testInitWithMainInPackage")
+    public void testInitOnExistingProjectWithNewPackage() throws Exception {
+        Path projectPath = tempProjectDirectory.resolve("firstTestWithPackagesMain");
+
+        sourceFiles = new ArrayList<>();
+        packageDescFiles = new ArrayList<>();
+
+        sourceFiles.add(new SrcFile("newpkg", FileType.MAIN));
+        packageDescFiles.add(new PackageMdFile("newpkg", FileType.MAIN));
+
+        InitHandler.initialize(projectPath, manifest, sourceFiles, packageDescFiles);
+
+        Assert.assertEquals(Files.exists(projectPath.resolve("newpkg").resolve("main.bal")), true);
```

Comment:
We can use `Assert.assertTrue()` function here.

Reply:
Fixed with 6ff8577